### PR TITLE
Build opencv winpack dldt

### PR DESCRIPTION
**Pull request by alalek**
Build package with OpenCV and [DLDT](https://github.com/opencv/dldt).

Details are below...<cut/>

Build prerequisites:
- Python 3.5+
- CMake
- Git Bash
- MSVS 2019 ([buildtools](https://aka.ms/buildtools) is enough)

Launch prerequisites:
- package itself requires MSVS 2019 redistibutables: [vc_redist.x64.exe](https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads)
- AVX2 CPU

Build steps:
- create build directory (~10 Gb is required)
- from build directory run `python <opencv_src>/platforms/winpack_dldt/build_package.py`
- check `package` directory
- sanity test:
```
.\package\opencv\build\setup_vars_opencv3.cmd python .\package\opencv\src\opencv\platforms\winpack_dldt\package-tests\test_dnn_backends.py
```

Notes:
- TBB parallel backend is used (from IE)

TODO:
- [x] verify package contents
- [ ] verify applications build with MSVS 2019 and older (MSVS 2015 / 2017)
- [x] verify debug process (of release binaries)
- [ ] check samples "run" scripts
- [x] add opencv .PDBs
- [x] `--build_test`
- [ ] SKIP: package dldt sources?
- [x] backlog: debug binaries?
- [ ] drop 32-bit ffmpeg binary
- [ ] SKIP: drop preview/debug tbb files

```
**WIP**
force_builders=Docs
build_image:Custom Win=winpack-dldt-2020.1-debug
```

### CODE REVIEW

File: platforms/winpack_dldt/build_package.py [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+        copy_bin('cache.json')
+        copy_dll('clDNNPlugin')
+        copy_dll('HeteroPlugin')
+        copy_dll('inference_engine')
+        copy_dll('inference_engine_nn_builder')
+        copy_dll('MKLDNNPlugin')
+        copy_dll('myriadPlugin')
+        copy_dll('ngraph')
+        copy_bin('plugins.xml')
+        copytree_smart(self.build_dir / 'bin' / 'intel64' / 'pcie-ma248x.elf', sysroot_bin_dir / 'pcie-ma248x.elf')
```
Comment (dkurt) [inline platforms/winpack_dldt/build_package.py]: In the distribution this file named as `pcie-ma248x.mvcmd`

File: platforms/winpack_dldt/build_package.py [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+
+        log.info('DLDT sysroot preparation completed')
+
+
+#===================================================================================================
+
+class Builder:
+    def __init__(self, config):
+        self.config = config
+        self.build_dir = prepare_dir(Path(self.config.output_dir) / 'opencv_build', create=True)
```
Comment (mshabunin) [inline platforms/winpack_dldt/build_package.py]: Shouln't `build_dir` be located in `.build_cache`?

File: platforms/winpack_dldt/build_package.py [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+        self.package_dir = prepare_dir(Path(self.config.output_dir) / 'package/opencv', create=True)
```
Comment (mshabunin) [inline platforms/winpack_dldt/build_package.py]: Shouldn't `package_dir` be the same as `output_dir`? For now `output_dir` contains only `package` and `opencv_build`, if `opencv_build` will be moved to `.build_cache`, then whole hierarchy can be flattened (1-2 levels).

File: platforms/winpack_dldt/build_package.py [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+            OPENCV_CONFIG_INSTALL_PATH='cmake',
+            OPENCV_3P_LIB_INSTALL_PATH='3rdparty',
+            OPENCV_SAMPLES_SRC_INSTALL_PATH='samples',
+            OPENCV_DOC_INSTALL_PATH='doc',
+            OPENCV_OTHER_INSTALL_PATH='etc',
+            OPENCV_LICENSES_INSTALL_PATH='etc/licenses',
+
+            BUILD_opencv_python2='OFF',
+            BUILD_opencv_python3='ON',
+            PYTHON3_LIMITED_API='ON',
```
[the remark this answers is not in the document]
Comment (mshabunin) [reply platforms/winpack_dldt/build_package.py]: Apparently I have some problems with python loader:
```.txt
(build) C:\work\winpack>python
Python 3.7.0 | packaged by conda-forge | (default, Nov 12 2018, 20:47:31) [MSC v.1900 64 bit (AMD64)] :: Anaconda, Inc. on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import cv2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\work\winpack\package\opencv\build\python\cv2\__init__.py", line 96, in <module>
    bootstrap()
  File "C:\work\winpack\package\opencv\build\python\cv2\__init__.py", line 62, in bootstrap
    ], True)
  File "C:\work\winpack\package\opencv\build\python\cv2\__init__.py", line 56, in load_first_config
    raise ImportError('OpenCV loader: missing configuration file: {}. Check OpenCV installation.'.format(fnames))
ImportError: OpenCV loader: missing configuration file: ['config-3.7.py', 'config-3.py']. Check OpenCV installation.
```
Configuration:
```
--   Python 3:
--     Interpreter:                 C:/Miniconda3/envs/build/python.exe (ver 3.7)
--     Libraries:                   NO
--     numpy:                       C:/Miniconda3/envs/build/lib/site-packages/numpy/core/include (ver 1.18.1)
--     install path:                -
```

**Comment by alalek** [comment]
:+1:

**Pull request merged**